internal/ioctl/ioctl.go

```diff
@@ -161,6 +174,85 @@ func GetCBTInfo() ([]types.CBTInfo, error) {
 	return cbtInfoRet, nil
 }
 
+func getKernelEntryAddress(name string) (uint64, error) {
+	var address uint64
+	execPath, err := os.Executable()
+	if err != nil {
+		return address, errors.Wrap(err, "getting the path of the executable")
+	}
+	out, err := exec.Command("sudo", execPath, "-ksym-address", name).Output()
+	if err != nil {
+		return address, errors.Wrap(err, "running sudo command with flag -ksym-address")
+	}
+	address, err = strconv.ParseUint(string(out), 16, 64)
+	if err != nil {
+		return address, errors.Wrapf(err, "converting %s to uint64", string(out))
+	}
+
+	return address, nil
+}
+
+func setMissingKernelEntries(ioctl_device *os.File) error {
+	var unresolvedEntryName string
+	var buffer [4096]C.char
+	unresolvedEntries := C.struct_ioctl_get_unresolved_kernel_entries_s{
+		buf: buffer,
+	}
+	r1, _, errno := syscall.Syscall(syscall.SYS_IOCTL, ioctl_device.Fd(), IOCTL_GET_UNRESOLVED_KERNEL_ENTRIES, uintptr(unsafe.Pointer(&unresolvedEntries)))
+	if int(errno) != 0 {
+		return errors.Wrap(errno, "getting unresolved kernel entries")
+	} else if r1 == 0 {
+		log.Println("No unresolved kernel entries found!")
+		return nil
+	} else {
+		// NOTE: The IOCTL_GET_UNRESOLVED_KERNEL_ENTRIES call only saves one kernel entry in the return buffer, no need for word separation.
+		unresolvedEntryName = C.GoString(&unresolvedEntries.buf[0])
+		log.Printf("Found unresolved kernel entry: %s\n", unresolvedEntryName)
```

Comment:
Does this include the memory address? If so, please don't log.

Reply:
No, as the variable name says, it only contains the name of the kernel entry. That's the main and only reason we need to read from `/proc/kallsyms`, to get the address. Will make sure to not log any such address